### DISCUSSION TITLE
SelectPicker infinite scroll

### DISCUSSION THREAD

**Discussion by abdullaev388** [rsuite/rsuite#2627] [Q&A] [answered]
Hello. I am using SelectPicker. 
Is there any way to load more data items when dropdown list scrolls to the bottom.
I mean infinite loader.

**Comment by simonguo** [comment, marked as answer]
You might try listening to the `onScroll` event in [`listProps`](https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md#prop-types) to dynamically update the data.  

```tsx
const listProps = { onScroll: e => console.log(e) };

return <SelectPicker virtualized listProps={listProps} />;
```

But we do not recommend this, and we are still evaluating whether to replace react-virtualized.  If you have any good suggestions or solutions, you can tell us, or give us PR.

**Reply by abdullaev388** [reply]
the creator of react-virtualized has another lightweight library for virtualization, it is react-window:
https://react-window.vercel.app/#/examples/list/fixed-size

you can replace the react-virtualized list with react-window's FixedSizeList. As the author says it is even more recommended than react-virtualized.
Also there is another library called react-window-infinite-loader:
https://github.com/bvaughn/react-window-infinite-loader
This library provides interface for easy infinite-scroll implementation.

So i am not very experienced to create PR tbh, but i would like to see in SelectPicker new prop: infiniteLoaderProps or even another picker component which supports react-window's infinite loader.

Thank you!

**Reply by abdullaev388** [reply]
any news?

**Reply by simonguo** [reply]
Please follow the progress of [PR 2595](https://github.com/rsuite/rsuite/pull/2595).

**Reply by simonguo** [reply]
Example about infinite loader:  https://rsuite-nextjs-git-feat-react-18-rsuite.vercel.app/guide/performance/

**Reply by abdullaev388** [reply]
great. Thanks

**Reply by abdullaev388** [reply]
When this will be in main branch?
